lib/api/CDataFrameAnalysisSpecification.cc

```diff
@@ -70,82 +73,30 @@ CDataFrameAnalysisSpecification::CDataFrameAnalysisSpecification(const std::stri
-                HANDLE_FATAL(<< "Input error: missing '" << name << "' in analysis "
-                             << "specification. Please report this problem.");
-            }
-        };
-
-        if (document.HasMember(ROWS) && isPositiveInteger(document[ROWS])) {
-            m_NumberRows = document[ROWS].GetUint();
-        } else {
-            registerFailure(ROWS);
-        }
-        if (document.HasMember(COLS) && isPositiveInteger(document[COLS])) {
-            m_NumberColumns = document[COLS].GetUint();
-        } else {
-            registerFailure(COLS);
-        }
-        if (document.HasMember(MEMORY_LIMIT) && isPositiveInteger(document[MEMORY_LIMIT])) {
-            m_MemoryLimit = document[MEMORY_LIMIT].GetUint();
-        } else {
-            registerFailure(MEMORY_LIMIT);
-        }
-        if (document.HasMember(THREADS) && isPositiveInteger(document[THREADS])) {
-            m_NumberThreads = document[THREADS].GetUint();
-        } else {
-            registerFailure(THREADS);
-        }
-        // TODO Remove if (false) hack when being passed.
-        if (false) {
-            if (document.HasMember(TEMPORARY_DIRECTORY) &&
-                document[TEMPORARY_DIRECTORY].IsString() &&
-                boost::filesystem::portable_name(document[TEMPORARY_DIRECTORY].GetString())) {
-                m_TemporaryDirectory = document[TEMPORARY_DIRECTORY].GetString();
-            } else {
-                registerFailure(TEMPORARY_DIRECTORY);
-            }
-        }
+        auto parameters = CONFIG_READER.read(specification);
 
-        if (document.HasMember(ANALYSIS) && document[ANALYSIS].IsObject()) {
-            const auto& analysis = document[ANALYSIS];
-            if (analysis.HasMember(NAME) && analysis[NAME].IsString()) {
-                this->initializeRunner(analysis[NAME].GetString(), analysis);
-            } else {
-                registerFailure(NAME);
+        for (auto name : {ROWS, COLS, MEMORY_LIMIT, THREADS}) {
+            if (parameters[name].as<std::size_t>() == 0) {
+                HANDLE_FATAL(<< "Input error: '" << name << "' must be non-zero");
             }
-        } else {
-            registerFailure(ANALYSIS);
         }
-
-        // Check for any unrecognised fields; these might be typos.
-        for (auto i = document.MemberBegin(); i != document.MemberEnd(); ++i) {
-            if (isValidMember(*i) == false) {
-                HANDLE_FATAL(<< "Input error: unexpected member '" << i->name.GetString()
-                             << "' of analysis specification. Please report this problem.");
-            }
+        m_NumberRows = parameters[ROWS].as<std::size_t>();
+        m_NumberColumns = parameters[COLS].as<std::size_t>();
+        m_MemoryLimit = parameters[MEMORY_LIMIT].as<std::size_t>();
+        m_NumberThreads = parameters[THREADS].as<std::size_t>();
+        m_TemporaryDirectory = parameters[TEMPORARY_DIRECTORY].fallback(boost::filesystem::current_path().string());
```

Comment:
looks dangerous to me, probably better to enforce it to not accidentally write somewhere.

(FWIW for forecast it only leads to an error if tmp storage is required)

Reply:
I agree. I've left a TODO to make this a required parameter. We need to update the Java to pass this as soon as that change is made I'll make this change.